### PR TITLE
Feat(eos_cli_config_gen): Add schema for hardware_counters

### PR DESCRIPTION
**Pull request by ghost**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Tony
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/hardware_counters.schema.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  hardware_counters:
+    type: list
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/hardware_counters.schema.yml]: This should be a dict, with features as the only key.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/hardware_counters.schema.yml (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/hardware_counters.schema.yml]: Corrected!
 This one is in draft due to convert_types issue for list

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/hardware_counters.schema.yml]: added validator support

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/hardware_counters.schema.yml]: Thanks

**Review by ClausHolbechArista — APPROVED** [review]

**Review by tgodaA — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml [hunk 2]
```diff
@@ -1081,6 +1081,28 @@ keys:
...
+        secondary_key: direction
+        convert_types:
+        - list
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              required: true
+              display_name: Name of feature
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: I think, this was removed recently. Please fix this.

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: Fixed

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**